unified-codes/tools/scripts/dgraph/migration/v1/export.js

```diff
@@ -0,0 +1,190 @@
+const fs = require('fs').promises;
+const excel = require('exceljs');
+
+//const { parser } = require('./parser');
+//const jsonFormat = require('json-format');
+
+const inputFilename = process.argv[2] || './data/data.json';
+const outputFilename = process.argv[3] || './data/products.xlsx';
+const items = [];
+
+const parseFile = async (data) => {
+  const jsonData = JSON.parse(data);
+  const { set } = jsonData;
+  const parent = { form: '', formCategory: '', product: { code: '', description: '' } };
+
+  set.forEach((item) => parseProduct(item, parent));
+
+  return new Promise((resolve) => resolve());
+};
+
+const parseChildren = (children, parent) => {
+  if (children) {
+    children.forEach((child) => parseProduct(child, parent));
+  }
+};
+
+const getProperty = (properties, type) => {
+  if (!properties) return '';
+  const prop = properties.find((p) => p.type === type);
+  return prop ? prop.value : '';
+};
+
+const addPropertiesToProduct = (product, properties) => {
+  product.strength = getProperty(properties, 'strength');
+  product.ddd = getProperty(properties, 'ddd');
+  product.rxnav = getProperty(properties, 'code_rxnav');
+  product.eml = getProperty(properties, 'who_eml');
+  product.atc = getProperty(properties, 'code_who_atc');
+  product.unspsc = getProperty(properties, 'code_unspsc');
+  product.nzulm = getProperty(properties, 'code_nzulm');
+};
+
+const getProduct = (item, parent) => {
+  const { code, description, type } = item;
+  const product = {
+    code,
+    description,
+    type,
+    form: parent.form,
+    formCategory: parent.formCategory,
+    product: parent.product,
+  };
+  addPropertiesToProduct(product, item.has_property);
+  return product;
+};
+
+const parseProduct = (item, parent) => {
+  const { code, description, type } = item;
+  switch (item.type) {
+    case 'unit_of_use':
+      items.push(getProduct(item, parent));
+      break;
+    case 'other':
+    case 'drug':
+    case 'medicinal_product':
+      if (item.has_child) {
+        parent.product = { code, description, type };
+        addPropertiesToProduct(parent.product, item.has_property);
+        parseChildren(item.has_child, parent);
+      } else {
+        items.push(getProduct(item, parent));
+      }
+      break;
+    case 'form':
+      parent.form = description;
+      parseChildren(item.has_child, parent);
+      break;
+    case 'form_category':
+      parent.formCategory = description;
+      parseChildren(item.has_child, parent);
+      break;
+  }
+};
```

Comment:
Don't want to mess with the recursion here too much, but I find it's a bit easier when recursive functions just call themselves with parameters, e.g. instead of `parseProduct` and `parseChildren` can have a generic `parseEntity` which handles both cases (can do a `forEach` for the children, and can look at the properties of the `entity` passed as a param to see how to handle, maybe with a `case` statement or some `if` statements)?